core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -1693,20 +1690,13 @@ class PartitionTest extends AbstractPartitionTest {
     val replicas = List(brokerId, remoteBrokerId)
     val isr = Set(brokerId)
 
-    val metadataCache: MetadataCache = if (kraft) mock(classOf[KRaftMetadataCache]) else mock(classOf[ZkMetadataCache])
-    if (kraft) {
-      addBrokerEpochToMockMetadataCache(metadataCache.asInstanceOf[KRaftMetadataCache], replicas)
-    }
+    val metadataCache: MetadataCache = mock(classOf[KRaftMetadataCache])
```

Comment:
We can use `KRaftMetadataCache` as the type so we don't need the multiple `asInstanceOf` calls later